### ISSUE TITLE
OBS on Flatpak "Failed to open NVENC codec: Uknown error occured"

### ISSUE THREAD

**Issue by BinaryVigilante** [obsproject/obs-studio#10433]
### Operating System Info

Other

### Other OS

Arch Linux

### OBS Studio Version

30.1.1

### OBS Studio Log URL

https://obsproject.com/logs/uOLC6vNl2jFqPR4C

### Expected Behavior

.

### Current Behavior

.

### Steps to Reproduce

.

### Anything else we should know?

I have the latest OBS version of 30.1.1 installed on my Arch Linux system using Flatpak. OBS randomly stops working between my machine being powered down, without any updates occurring to either Nvidia nor OBS. The log file says one thing, and from what I can see the terminal output says something more detailed:

```
warning: [NVENC encoder] nvenc_create_internal failed, trying again without Psycho Visual Tuning
info: ---------------------------------
info: [FFmpeg NVENC encoder: 'advanced_video_recording'] settings:
	encoder:      NVIDIA NVENC H.264 (FFmpeg)
	rate_control: CQP
	bitrate:      0
	cqp:          16
	keyint:       -1
	preset:       p6
	tuning:       hq
	multipass:    disabled
	profile:      high
	width:        1920
	height:       1080
	b-frames:     2
	psycho-aq:    0
	GPU:          0

[h264_nvenc @ 0x636d6d9845c0] dl_fn->cuda_dl->cuInit(0) failed -> CUDA_ERROR_UNKNOWN: unknown error
```

Again Nvidia hasn't been updated in-between power-downs nor has the Kernel or OBS.

**Comment by BinaryVigilante** [comment]
Also just as a note not related to that issue, it would be really nice if it wasn't forced to use the given template for bug reports. Not all reports fit that template and can be put into that structure :)

**Comment by BinaryVigilante** [comment]
Update: Randomly started working again after I used these commands as root:
`rmmod nvidia_uvm`
`modprobe nvidia_uvm`

I say randomly as I tried them previously and it didn't an effect. This is also not a viable solution to the issue, just something to point out.

**Comment by BinaryVigilante** [comment]
> We do not use GitHub Issues for support questions.
> 
> Please use the [Discord server](https://obsproject.com/discord) or the [forums](https://obsproject.com/forum). For more info, check our [Help page](https://obsproject.com/help).

This is not a support question? OBS literally doesn't work, and provides an error in the terminal output.

**Comment by Fenrirthviti** [comment]
This was an environment issue, not a bug, which falls under support.

**Comment by BinaryVigilante** [comment]
I'm sorry, but if I may ask, how do you know it's an environment issue and not a bug, as it's OBS giving the error and not Flatpak?

**Comment by derrod** [comment]
> I'm sorry, but if I may ask, how do you know it's an environment issue and not a bug, as it's OBS giving the error and not Flatpak?

The error is not in OBS but in CUDA, which part of the NVIDIA driver, which is provided through a Flatpak runtime.

**Comment by AndreCox** [comment]
> Update: Randomly started working again after I used these commands as root: `rmmod nvidia_uvm` `modprobe nvidia_uvm`
> 
> I say randomly as I tried them previously and it didn't have an effect. This is also not a viable solution to the issue, just something to point out.

Thank you for posting this fix, it worked for me. Quite annoying that I had to spend days getting a "basic" feature working.

**Comment by derrod** [comment]
This seems to be a known issue with NVIDIA drivers when resuming from sleep, which is outside of our control: https://forums.developer.nvidia.com/t/bug-nvidia-uvm-needs-to-be-removed-and-re-inserted-in-order-to-work-after-wakeup-from-suspend/188453

**Comment by BinaryVigilante** [comment]
> This seems to be a known issue with NVIDIA drivers when resuming from sleep, which is outside of our control: https://forums.developer.nvidia.com/t/bug-nvidia-uvm-needs-to-be-removed-and-re-inserted-in-order-to-work-after-wakeup-from-suspend/188453

I don't think that this is the reason, as even with using the "fix" I posted further up, it hasn't worked and have been unable to get it to work after it broke again.

> The error is not in OBS but in CUDA, which part of the NVIDIA driver, which is provided through a Flatpak runtime.

How do you know it's an error in CUDA and not in OBS?

> This was an environment issue, not a bug, which falls under support.

Yes it is provided through a Flatpak runtime, but it is OBS that is officially providing OBS through Flatpak, so clearly, there is an **issue** with OBS on Flatpak that needs looking into.

This is very much still an issue, and it has not been solved yet. Please re-open this report.

**Comment by Fenrirthviti** [comment]
As explained above, this is a known issue with NVIDIA's drivers. There's nothing for us to fix.

**Comment by BinaryVigilante** [comment]
Oh my bad, I took a closer look at the link. I'm not sure if that is the cause of the issue, but I guess we will see when/if that bug is resolved.

I have however tried what other people has as well, using `modprobe` without success. Also my PC never sleeps, so I'm not sure if that issue is related.

**Comment by BinaryVigilante** [comment]
Is there anything I can do to provide further logs, information or something similar that might shed more light on the issue?

**Comment by derrod** [comment]
> > The error is not in OBS but in CUDA, which part of the NVIDIA driver, which is provided through a Flatpak runtime.
> 
> How do you know it's an error in CUDA and not in OBS?

See the log:
`[h264_nvenc @ 0x636d6d9845c0] dl_fn->cuda_dl->cuInit(0) failed -> CUDA_ERROR_UNKNOWN: unknown error`

Furthermore, this error comes from NVENC in FFmpeg, not OBS. So it's doubly outside of our control.
 
> Yes it is provided through a Flatpak runtime, but it is OBS that is officially providing OBS through Flatpak, so clearly, there is an **issue** with OBS on Flatpak that needs looking into.

By that logic we're responsible for Microsoft's bugs on Windows because we ship on Windows.
